### ISSUE TITLE
$request->hasSession() is always true

### ISSUE THREAD

**Issue by VincentLanglet** [phpstan/phpstan-symfony#180]
Hi @ondrejmirtes.

The following code is reported as an error by phpstan-strict-rules
```
/**
     * @param Request $request
     *
     * @return array
     */
    public function getCredentials(Request $request)
    {
        $credentials = [
            'username'   => $request->request->get('username'),
            'password'   => $request->request->get('password'),
            'csrf_token' => $request->request->get('_csrf_token'),
        ];

        if ($request->hasSession()) {
            $request->getSession()->set(Security::LAST_USERNAME, $credentials['username']);
        }

        return $credentials;
    }
```

> Call to method Symfony\Component\HttpFoundation\Request::hasSession() will always evaluate to true.

I don't know why and didn't succeed to reproduce it in the playground.

the method hasSession is 
```
/**
     * Whether the request contains a Session object.
     *
     * This method does not give any information about the state of the session object,
     * like whether the session is started or not. It is just a way to check if this Request
     * is associated with a Session instance.
     *
     * @return bool true when the Request contains a Session object, false otherwise
     */
    public function hasSession()
    {
        return null !== $this->session;
    }
```

The property in the request has wrong phpdoc
```
    /**
     * @var SessionInterface|callable
     */
    protected $session;
```

I tried removing the `true when the Request contains a Session object, false otherwise` part of the phpdoc or adding `null` to the protected property, but it changed nothing (even if I was hoping for phpstan to only look at the `@return bool` part).

Any idea why the method is considered as always returning true ? Is it a bug ?

**Comment by ondrejmirtes** [comment]
Duplicate of https://github.com/phpstan/phpstan-symfony/issues/86
